File: game_plugin_interface/src/sc/protocol/responses/GamePausedEvent.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;

import sc.api.plugins.IPlayer;

@XStreamAlias(value="paused")
public class GamePausedEvent
{
	private final IPlayer	nextPlayer;

	public GamePausedEvent(IPlayer nextPlayer)
	{
		this.nextPlayer = nextPlayer;
	}

	public IPlayer getNextPlayer()
	{
		return this.nextPlayer;
	}
}

File: sdk/src/server-api/sc/protocol/responses/GamePausedEvent.kt
package sc.protocol.responses

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

import sc.framework.plugins.Player
import sc.protocol.RoomMessage

/**
 * Indicates that the game has been paused.
 *
 * @param nextPlayer the next Player to move after unpausing
 */
@XStreamAlias(value = "paused")
data class GamePausedEvent(
        @XStreamAsAttribute
        val nextPlayer: Player
): RoomMessage
